File: 202409/58.length-of-last-word.java
/*
 * @lc app=leetcode id=58 lang=java
 *
 * [58] Length of Last Word
 */

// @lc code=start
class Solution {
    public int lengthOfLastWord(String s) {
        // return s.trim().length() - s.trim().lastIndexOf(" ") - 1; // wtf?
        int len = s.length();
        // iterate from the end of the string
        int end = len - 1;
        int start = 0;
        // find the first non-space character from the end
        while (end >= 0 && s.charAt(end) == ' ') {
            end--;
        }

        // find the first space character from the end
        start = end;
        while (start >= 0 && s.charAt(start) != ' ') {
            start--;
        }
        return end - start;
    }
}
// @lc code=end

class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.lengthOfLastWord("Hello World"));
    }
}

File: 202409/58.length-of-last-word.cpp
/*
 * @lc app=leetcode id=58 lang=cpp
 *
 * [58] Length of Last Word
 */

#include <iostream>
#include <string>
using namespace std;

// @lc code=start
class Solution
{
public:
    int lengthOfLastWord(string s)
    {
        auto riter1 = s.rbegin();
        auto riter2 = s.rbegin();
        while (riter1 != s.rend() && *riter1 == ' ')
        {

            riter1++;
        }
        riter2 = riter1;
        while (riter2 != s.rend() && *riter2 != ' ')
        {
            riter2++;
        }

        return riter2 - riter1;
    }
};
// @lc code=end

int main()
{
    Solution s;
    cout << s.lengthOfLastWord("Hello Worfld  ") << endl;
    return 0;
}